whisper/whisperv5/peer.go

```diff
@@ -113,7 +117,27 @@ func (p *Peer) update() {
 				log.Trace("broadcast failed", "reason", err, "peer", p.ID())
 				return
 			}
-
+		case <-hashesTransmit.C:
+			if err := p.broadcastHashes(hashes); err != nil {
+				log.Trace("broadcast of hashes failed", err, "peer", p.ID)
+				return
+			}
+			hashes = hashes[:0]
+		case hash := <-p.hashes:
+			if _, ok := p.advertised[hash]; ok {
+				continue
+			}
+			if p.known.Has(hash) {
+				continue
+			}
+			hashes = append(hashes, hash)
+			if len(hashes) == cap(hashes) {
```

Comment:
One potential reason for my question in the general comment is that you wait too long to send the hashes, resulting in duplicates. I would go for half the size and see if it improves. I can imagine there is a trade-off between sending duplicates and waiting for hashes, and I doubt we're at the break-even point.

Reply:
hashes are sent once every 100ms or when capacity is reached